integration/specs/CodeInput/codeInput-1.spec.js

```diff
@@ -0,0 +1,107 @@
+const PageCodeInput = require('../../../src/components/CodeInput/pageObject');
+const { DELETE_KEY, TAB_KEY } = require('../../constants');
+
+const CODEINPUT = '#codeinput-1';
+
+describe('CodeInput base example', () => {
+    beforeAll(() => {
+        browser.url('/#!/CodeInput/1');
+    });
+    beforeEach(() => {
+        browser.refresh();
+        const component = $(CODEINPUT);
+        component.waitForExist();
+    });
+    it('should have focus on the first input on load', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+    });
+    it('should change focus to the second input when the first input its filled with a number', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getFocusedIndex()).toBe(1);
+    });
+    it('should remove value from first input when first input is selected, has a value and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getInputValueByIndex(0)).toBe('2');
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+    });
+    it('should return focus to the first input and clear his value when the first input has a value, the second input is empty, is focused and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        expect(codeInput.getInputValueByIndex(0)).toBe('2');
+        expect(codeInput.getFocusedIndex()).toBe(1);
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+    });
+    it('should keep focus on the first input when input is empty and "Backspace" key is pressed', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        expect(codeInput.getInputValueByIndex(0)).toBe('');
+        browser.keys(DELETE_KEY);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+    });
+    it('should keep focus on the last input when the last input value is filled with a number', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        browser.keys(['', '2']);
+        expect(codeInput.getFocusedIndex()).toBe(3);
+    });
+    it('should keep focus in the current input when we type a letter', () => {
+        const codeInput = new PageCodeInput(CODEINPUT);
+        expect(codeInput.getFocusedIndex()).toBe(0);
+        browser.keys(['', 'A']);
+        expect(codeInput.getFocusedIndex()).toBe(0);
```

Comment:
add another test like this that check for the input to be empty in order to prove that it does not accepts other values than numbers